### ISSUE TITLE
Validation for the manifest.json file reports missing files when those files are ignored

### ISSUE THREAD

**Issue by rpl** [mozilla/addons-linter#2269]
### Describe the problem and steps to reproduce it:

The validation for the manifest.json file reports existing files as missing (e.g. MANIFEST_BACKGROUND_FILE_NOT_FOUND) if those fails are being ignored (based on the shouldScanFile option passed by web-ext lint: https://github.com/mozilla/web-ext/blob/7f90caa28aa08421879e41fe04049d40e48934b6/src/cmd/lint.js#L94).

The following STR can be used to reproduce the issue:
- clone the https://github.com/mozilla/email-tabs repository
- enter the email-tabs directory and run `npm install` and `npm run build`
- validate the package using web-ext lint with the --ignore-files option:
  ```web-ext lint -s addon --ignore-files='build/*.js' --self-hosted```
(Please include as many details as possible.)

### What happened?

The following validation warnings have been produced:
```
Code                                 Message                   Description                                         File            Line   Column
MANIFEST_BACKGROUND_FILE_NOT_FOUND   A background script       Background script could not be found at             manifest.json      
                                     defined in the manifest   "build/testpilot-ga.js".                                               
                                     could not be found.
...
```

### What did you expect to happen?

The validation on the manifest.json file should still be able to detect that the files exists, even if the developer doesn't want the linter to produce actual linting errors related to those files.

### Anything else we should know?

This issue as been originally reported on the web-ext repo as: mozilla/web-ext#1376

**Comment by pdehaan** [comment]
I just encountered this today while trying to ignore an included vendor file in our add-on, except [interestingly] now this is an **error** and not a **warning**:

```sh
$ npx web-ext lint --self-hosted --ignore-files=vendor/fxa-crypto-relier.js

Applying config files: ./package.json, ./web-ext-config.js
Validation Summary:

errors          1
notices         0
warnings        14

ERRORS:

Code                                 Message                              Description                               File            Line   Column
MANIFEST_BACKGROUND_FILE_NOT_FOUND   A background script defined in the   Background script could not be found at   manifest.json
                                     manifest could not be found.         "vendor/fxa-crypto-relier.js".
WARNINGS:

Code                    Message                          Description                                                           File            Line   Column
UNSAFE_VAR_ASSIGNMENT   Unsafe assignment to innerHTML   Due to both security and performance concerns, this may not be set    view.js         34     7
                                                         using dynamic values which have not been adequately sanitized. This
                                                         can lead to security issues or fairly serious performance
                                                         degradation.

...
```

So in my case, our add-on is including a third party script, but I don't necessarily care about any code warnings or errors in that file, but I want to avoid the `MANIFEST_BACKGROUND_FILE_NOT_FOUND` since the file is there, I just want to suppress any linting errors/warnings from that particular file.

**Comment by ianb** [comment]
I'm still able to reproduce this issue.

**Comment by a13o** [comment]
Is there a workaround for this issue if you aren't using a js bundler?

Say you use npm to grab a vendor dependency, then include it in your manifest:
```json
"background": {
    "scripts": [
      "node_modules/webextension-polyfill/dist/browser-polyfill.min.js",
      "src/background.js"
    ]
}
```
(as per https://github.com/mozilla/webextension-polyfill#basic-setup)

`web-ext lint` will fail even if you can run the extension successfully, because node_modules is a default ignored folder.

Also MANIFEST_BACKGROUND_FILE_NOT_FOUND is reported as an error, not a warning. The P4 status may need to be revisited. This linter currently doesn't support my use case.
